bindata/network/ovn-kubernetes/ovnkube-node.yaml

```diff
@@ -28,77 +22,166 @@ spec:
-      # ovs flow for ovn (geneve)
-      # /usr/share/openvswitch/scripts/ovs-ctl --protocol=udp --dport=6081 enable-protocol
-      # The network container launches the ovn-k8s-cni-overlay process, the kube-proxy, and the local DNS service.
-      # It relies on an up to date node-config.yaml being present.
+        terminationMessagePolicy: FallbackToLogsOnError
+        livenessProbe:
+          exec:
+            command:
+            - /usr/share/openvswitch/scripts/ovs-ctl
+            - status
+          initialDelaySeconds: 15
+          periodSeconds: 5
+        readinessProbe:
+          exec:
+            command:
+            - /usr/share/openvswitch/scripts/ovs-ctl
+            - status
+          initialDelaySeconds: 15
+          periodSeconds: 5
+        lifecycle:
+          preStop:
+            exec:
+              command: ["/usr/share/openvswitch/scripts/ovs-ctl", "stop"]
+        terminationGracePeriodSeconds: 10
       - name: ovn-controller
         image: {{.OvnImage}}
-
-        command: ["/root/ovnkube.sh", "ovn-controller"]
-
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f "/env/${K8S_NODE}" ]]; then
+            set -o allexport
+            source "/env/${K8S_NODE}"
+            set +o allexport
+          fi
+          exec ovn-controller unix:/var/run/openvswitch/db.sock -vfile:off \
+            --no-chdir --pidfile=/var/run/openvswitch/ovn-controller.pid \
+            -vconsole:"${OVN_LOG_LEVEL}"
         securityContext:
-          # Permission could be reduced by selecting an appropriate SELinux policy
           privileged: true
-
+        env:
+        - name: OVN_LOG_LEVEL
+          value: info
+        - name: K8S_NODE
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
         volumeMounts:
-        # Directory which contains the host configuration.
-        - mountPath: /var/lib/openvswitch/
-          name: host-var-lib-ovs
-        - mountPath: /var/run/openvswitch/
-          name: host-var-run-ovs
-
+        - mountPath: /run/openvswitch
+          name: host-run-openvswitch
+        - mountPath: /etc/openvswitch
+          name: host-var-lib-openvswitch
+        - mountPath: /var/lib/openvswitch
+          name: host-var-lib-openvswitch
+        - mountPath: /env
+          name: env-overrides
+        terminationMessagePolicy: FallbackToLogsOnError
         resources:
           requests:
             cpu: 100m
             memory: 300Mi
+      - name: ovnkube-node
+        image: {{.OvnImage}}
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f "/env/${K8S_NODE}" ]]; then
+            set -o allexport
+            source "/env/${K8S_NODE}"
+            set +o allexport
+          fi
+          cp -f /usr/libexec/cni/ovn-k8s-cni-overlay /cni-bin-dir/
+          db_ip=$(kubectl get ep ovnkube-db -o jsonpath='{.subsets[0].addresses[0].ip}')
```

Comment:
You're assuming this already exists, which is not at all guaranteed

Reply:
if not, it will fail, and the process will crashloop until it does.

Reply:
I feel like crashlooping should not be an expected part of startup.

Reply:
True enough, though this will all go away when we have HA, and endpoints are just watched automatically.

Reply:
Added a retry loop.